### DISCUSSION TITLE
Possibility of forced alignment support?

### DISCUSSION THREAD

**Discussion by conncomg123** [openai/whisper#52] [Ideas]
This already supports timestamps for phrases-- would it be possible for forced alignment support? That is, given an audio file and, optionally, its transcript, could it produce a file containing the timestamps of each phoneme?

**Comment by jongwook** [comment]
Discussion #3 has some implementations for getting token/word-level timestamps. Getting timestamps for each phoneme would be difficult from Whisper models only, because the model is end-to-end trained to predict BPE tokens directly, which are often a full word or subword consisting of a few graphemes.

An alternative could be using an external forced alignment tool based on the outputs from a Whisper model.

**Comment by jongwook** [comment]
Update: I've made a demo of obtaining word-level timestamps using the cross-attention patterns in [the multilingual ASR notebook](https://github.com/openai/whisper/blob/main/notebooks/Multilingual_ASR.ipynb)

![image](https://user-images.githubusercontent.com/266841/210467619-4384e3d0-522b-4e84-8c64-60e6422eed26.png)
